src/parser/expression.js

```diff
@@ -551,7 +551,7 @@ pp.parseParenAndDistinguishExpression = function (startPos, startLoc, canBeArrow
     if (first) {
       first = false;
     } else {
-      this.expect(tt.comma);
+      this.expect(tt.comma, refNeedsArrowPos.start || null);
```

Comment:
basically this is doing: `refNeedsArrowPos.start > 0 ? refNeedsArrowPos.start : null`
This line is first triggered in the second iteration `if (first) ...` and `refNeedsArrowPos` is supplied to `parseMaybeAssign` and might be changed there.